src/main/java/org/jfrog/hudson/pipeline/types/deployers/Deployer.java

```diff
@@ -42,6 +44,8 @@
     private Filter artifactDeploymentPatterns = new Filter();
     private String customBuildName = "";
     private transient CpsScript cpsScript;
+    String snapshotRepo;
+    String releaseRepo;
```

Comment:
Since the *snapshotRepo* and *releaseRepo* members are applicable only for the *MavenDeployer*, let's see if we can them (and their relevant logic) to the *MavenDeployer* class.
I can see that these variables have actually been moved our of the *MavenDeployer* class and into this class  in this PR.